examples/cbf2adscimg_sub.c:
  warning: for loop has empty body

examples/changtestcompression.c:
  warning: implicit conversion from 'unsigned long' to 'double'
  changes value from 18446744073709551615 to 18446744073709551616

examples/dps_peaksearch.c:
  warning: a function definition without a prototype is deprecated in
  all versions of C and is not supported in C2x

examples/testcell.C:
  warning: using directive refers to implicitly-defined namespace
  'std'

examples/tif_sprint.c:
  warning: format string is not a string literal (potentially
  insecure)

examples/cbf_testxfelread.c:
  warning: if statement has empty body
--- a/examples/cbf2adscimg_sub.c
+++ b/examples/cbf2adscimg_sub.c
@@ -154,7 +154,8 @@
 /*
  * find the } marking the end of the information in the header
  */
-  for ( hp=header; *hp != '}'; hp++);
+  for ( hp=header; *hp != '}'; hp++)
+    ;
 
 /*
  * Write the field name starting at the position of the }
--- a/examples/cbf_testxfelread.c
+++ b/examples/cbf_testxfelread.c
@@ -1030,7 +1030,8 @@
         for (ii=0; ii < XFEL_DIAMETER; ii++) {
             for (jj=0; jj < XFEL_DIAMETER; jj++) {
                 if (outarray[ii][jj] != XFEL_UNDEFINED
-                    && outarray[ii][jj] != XFEL_OVERLOAD);
+                    && outarray[ii][jj] != XFEL_OVERLOAD)
+                        ;
                 intoutarray[ii][jj] = (int)(outarray[ii][jj]+0.5);
             }
         }
--- a/examples/changtestcompression.c
+++ b/examples/changtestcompression.c
@@ -120,7 +120,7 @@
     uldata = (unsigned long *)calloc(nelem, sizeof(long));
     elsize = sizeof(long);
     while (i < nelem) {
-      uldata[i] = urand() * ULONG_MAX;
+      uldata[i] = urand() * (double)ULONG_MAX;
       i += 1 + urand() * NSTEPS;
     }
     *data = uldata;
@@ -129,7 +129,7 @@
     sldata = (signed long *)calloc(nelem, sizeof(long));
     elsize = sizeof(long);
     while (i < nelem) {
-      sldata[i] = urand() * LONG_MAX;
+      sldata[i] = urand() * (double)LONG_MAX;
       i += 1 + urand() * NSTEPS;
     }
     *data = sldata;
@@ -139,7 +139,7 @@
     ulldata = (unsigned long long *)calloc(nelem, sizeof(long long));
     elsize = sizeof(long long);
     while (i < nelem) {
-      ulldata[i] = urand() * ULLONG_MAX;
+      ulldata[i] = urand() * (double)ULLONG_MAX;
       i += 1 + urand() * NSTEPS;
     }
     *data = ulldata;
@@ -148,7 +148,7 @@
     slldata = (signed long long *)calloc(nelem, sizeof(long long));
     elsize = sizeof(long long);
     while (i < nelem) {
-      slldata[i] = urand() * LLONG_MAX;
+      slldata[i] = urand() * (double)LLONG_MAX;
       i += 1 + urand() * NSTEPS;
     }
     *data = slldata;
--- a/examples/dps_peaksearch.c
+++ b/examples/dps_peaksearch.c
@@ -139,7 +139,7 @@
  *
  */
 
-void
+static void
 cmass (idata, iwidth, iheight, x, y, cm_x, cm_y, cm_nx, cm_ny)
 register unsigned short  *idata;
 int iwidth, iheight, x, y;
--- a/examples/testcell.C
+++ b/examples/testcell.C
@@ -185,8 +185,8 @@
 
 
 
-using namespace std;
 #include <iostream>
+using namespace std;
 #include "cbf.h"
 #include "cbf_simple.h"
 
--- a/examples/tif_sprint.c
+++ b/examples/tif_sprint.c
@@ -451,7 +451,7 @@
 		     cp = strchr(cp,'\0')+1, i--) {
 			size_t max_chars = 
 				inknameslen - (cp - inknames);
-            chars_used += snprintf(str+chars_used, ((xstrlen>chars_used)?xstrlen-chars_used:0),sep);
+            chars_used += snprintf(str+chars_used, ((xstrlen>chars_used)?xstrlen-chars_used:0), "%s", sep);
             chars_used += _TIFFsnprintAsciiBounded(str+chars_used, ((xstrlen>chars_used)?xstrlen-chars_used:0), cp, max_chars);
 			sep = ", ";
 		}
